File: CH2-modules/EX3-how-much-insurance/source-code/howMuchInsurance.js
const input = require("prompt-sync")();

function inputReplacementCost(replacementCost) {
  replacementCost.value = Number(input("Enter a replacement cost : $"));
}
function calculateInsurance(totalInsurance, replacementCost, minimumInsurance) {
  totalInsurance.value =
    replacementCost.value * minimumInsurance.value + replacementCost.value;
}
function printTotalInsurance(totalInsurance) {
  console.log("Minimum amount insurance : $" + totalInsurance.value);
}

function main() {
  let replacementCost = {
    value: 0,
  };
  let minimumInsurance = {
    value: 0.8,
  };
  let totalInsurance = {
    value: 0,
  };

  inputReplacementCost(replacementCost);
  calculateInsurance(totalInsurance, replacementCost, minimumInsurance);
  printTotalInsurance(totalInsurance);
}

// entry point
main();

File: CH2-modules/EX3-how-much-insurance/source-code/howMuchInsurance.cpp
#include <iostream>
using namespace std;

void inputReplacementCost(double &replacementCost)
{
    cout << "Enter a replacement cost : $";
    cin >> replacementCost;
}

void calculateInsurance(double &totalInsurance, double &replacementCost, double &minimumInsurance)
{
    totalInsurance = (replacementCost * minimumInsurance) + replacementCost;
}

void printTotalInsurance(double &totalInsurance)
{

    cout << "Minimum amount insurance : $" << totalInsurance;
}

int main()
{

    double replacementCost;
    double minimumInsurance = 0.8;
    double totalInsurance;

    inputReplacementCost(replacementCost);
    calculateInsurance(totalInsurance, replacementCost, minimumInsurance);
    printTotalInsurance(totalInsurance);
}